src/machine/usb_nrf52840.go

```diff
@@ -12,22 +12,100 @@ import (
 
 // USBCDC is the USB CDC aka serial over USB interface on the nRF52840
 type USBCDC struct {
-	Buffer       *RingBuffer
-	interrupt    interrupt.Interrupt
-	initcomplete bool
+	Buffer            *RingBuffer
+	interrupt         interrupt.Interrupt
+	initcomplete      bool
+	TxIdx             volatile.Register8
+	waitTxc           bool
+	waitTxcRetryCount uint8
+	sent              bool
+}
+
+const (
+	usbcdcTxSizeMask          uint8 = 0x3F
+	usbcdcTxBankMask          uint8 = ^usbcdcTxSizeMask
+	usbcdcTxBank1st           uint8 = 0x00
+	usbcdcTxBank2nd           uint8 = usbcdcTxSizeMask + 1
+	usbcdcTxMaxRetriesAllowed uint8 = 5
+)
+
+// Flush flushes buffered data.
+func (usbcdc *USBCDC) Flush() error {
+	if usbLineInfo.lineState > 0 {
+		idx := usbcdc.TxIdx.Get()
+		sz := idx & usbcdcTxSizeMask
+		bk := idx & usbcdcTxBankMask
+		if 0 < sz {
+
+			if usbcdc.waitTxc {
+				// waiting for the next flush(), because the transmission is not complete
+				usbcdc.waitTxcRetryCount++
```

Comment:
Unlike samd2x/samd5x, I decided to increment the `waitTxcRetryCount` here.
I think it would be better to unify the samd code as well.

Reply:
Good idea.